persistence/jdbc/src/test/java/org/infinispan/persistence/jdbc/binary/JdbcBinaryStoreTest.java

```diff
@@ -98,6 +99,18 @@ public void testPurgeExpiredAllCodepaths() throws Exception {
       UnitTestDatabaseManager.verifyConnectionLeaks(((JdbcBinaryStore) cl).getConnectionFactory());
    }
 
+   public void testCacheSize() {
+      long lifespan = 3000;
+      for (int i = 0; i < 120; i++) {
+         cl.write(marshalledEntry(internalCacheEntry("mortal" + i, "v1", lifespan)));
+         cl.write(marshalledEntry(internalCacheEntry("immortal" + i, "v1", -1)));
+      }
```

Comment:
@ryanemerson The test fails if I populate the entries like this:

      for (int i = 0; i < 240; i++) {
         if (i % 2 == 0) {
            cl.write(marshalledEntry(internalCacheEntry("key" + i, "v1", lifespan)));
         } else {
            cl.write(marshalledEntry(internalCacheEntry("key" + i, "v1", -1)));
         }
      }

Your test works because the mortal and immortal keys are hashed to different buckets, but if you mix mortal and immortal entries in the same bucket your count will be wrong. The same happens with different lifespans in the same bucket.